apps/rxjs-race-condition/src/app/topic.service.ts

```diff
@@ -1,10 +1,29 @@
 import { Injectable } from '@angular/core';
-import { map, timer } from 'rxjs';
+import { BehaviorSubject, Observable, map, timer } from 'rxjs';
 
 export type TopicType = 'Politic' | 'Sport' | 'Culture' | 'Nature';
 
 @Injectable({ providedIn: 'root' })
 export class TopicService {
+  topics$ = new BehaviorSubject<{ topics: TopicType[] }>({ topics: [] });
+
+  constructor() {
+    this.init();
+  }
+
   fakeGetHttpTopic = () =>
     timer(1000).pipe(map((): TopicType[] => ['Politic', 'Culture', 'Nature']));
+
+  init() {
+    this.fakeGetHttpTopic().subscribe((topics) => {
+      const data = {
+        topics: topics,
```

Comment:
nit, when the property and the value have the same name, you don't have to repeat it.